src/lib.rs

```diff
@@ -145,6 +148,25 @@ pub struct Node {
     pub id: String,
     /// List of opaque identifiers for this node's dependencies
     pub dependencies: Vec<String>,
+
+    /// Dependencies in a structured format
+    #[serde(default)]
+    pub deps: Vec<NodeDep>,
+    /// Features provided by the crate
+    #[serde(default)]
+    pub features: Vec<String>,
+    #[doc(hidden)]
+    #[serde(skip)]
+    __do_not_match_exhaustively: (),
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+/// A dependency in a node
+pub struct NodeDep {
+    /// Crate name
```

Comment:
For renamed dependencies, this is the new (renamed) name, not the original name.